src/arkode/arkode_lsrkstep_io.c

```diff
@@ -0,0 +1,661 @@
+
+  This set routine must be called after calling LSRKStepSetMethod with an SSP method
+  ---------------------------------------------------------------*/
+int LSRKStepSetSSPStageNum(void* arkode_mem, int num_of_stages)
+{
+  ARKodeMem ark_mem;
+  ARKodeLSRKStepMem step_mem;
+  int retval;
+
+  /* access ARKodeMem and ARKodeLSRKStepMem structures */
+  retval = lsrkStep_AccessARKODEStepMem(arkode_mem, __func__, &ark_mem,
+                                        &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  switch (step_mem->LSRKmethod)
+  {
+  case ARKODE_LSRK_SSP_S_2:
+    if (num_of_stages < 2)
+    {
+      arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                      "num_of_stages must be greater than or equal to 2");
+      return (ARK_ILL_INPUT);
+    }
+    break;
+
+  case ARKODE_LSRK_SSP_S_3:
+    if (num_of_stages < 9 || (SUNRceil(SUNRsqrt(num_of_stages)) !=
+                              SUNRfloor(SUNRsqrt(num_of_stages))))
+    {
+      arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__,
+                      __FILE__, "num_of_stages must be a perfect square greater than or equal to 9");
+      return (ARK_ILL_INPUT);
+    }
+    break;
+
+  case ARKODE_LSRK_SSP_10_4:
+    if (num_of_stages != 10)
+    {
+      arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                      "SSP10_4 method has a prefixed num_of_stages = 10");
+      return (ARK_ILL_INPUT);
+    }
+    break;
+
+  default:
+    arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    "Call LSRKStepSetMethod to declare SSP method type first!");
+    return (ARK_ILL_INPUT);
+    break;
+  }
+  step_mem->req_stages = num_of_stages;
+
+  return (ARK_SUCCESS);
+}
+
+/*===============================================================
+  Exported optional output functions.
+  ===============================================================*/
+
+/*---------------------------------------------------------------
+  LSRKStepGetNumRhsEvals:
+
+  Returns the current number of calls to f
+  ---------------------------------------------------------------*/
+int LSRKStepGetNumRhsEvals(void* arkode_mem, int num_rhs_fn, long int* f_evals)
+{
+  ARKodeMem ark_mem;
+  ARKodeLSRKStepMem step_mem;
+  int retval;
+
+  /* access ARKodeMem and ARKodeLSRKStepMem structures */
+  retval = lsrkStep_AccessARKODEStepMem(arkode_mem, __func__, &ark_mem,
+                                        &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  /* get values from step_mem */
+  *f_evals = step_mem->nfe;
+
+  return (ARK_SUCCESS);
+}
+
+/*---------------------------------------------------------------
+  LSRKStepGetNumDomEigUpdates:
+
+  Returns the number of dominant eigenvalue updates
+  ---------------------------------------------------------------*/
+int LSRKStepGetNumDomEigUpdates(void* arkode_mem, long int* num_dom_eig_updates)
```

Comment:
Check that `num_dom_eig_updates != NULL`

Reply:
We fail to do this check in the rest of ARKODE, but it would be good to add.  Could we just
```
SUNAssert(num_dom_eig_updates, SUN_ERR_ARG_CORRUPT);
```
??

Reply:
I think `SUNAssert` would be good here. @balos1 anything I'm forgetting?